src/exts/message.py

```diff
@@ -199,10 +199,37 @@ async def on_thread_create(self, thread: interactions.Thread):
 
     @interactions.extension_component("close thread")
     async def _close_thread(self, ctx: interactions.ComponentContext):
+        components = ctx.message.components
+        for row in components:
+            for component in row.components:
+                component.disabled = True
+        await ctx.edit(components=components)
         await ctx.get_channel()
-        await ctx.send("Closing! Thank you for using our help system!")
+        await ctx.channel.send(
+            "Marking as solved! Thank you for using our help system!\n"
+            "If you wish to reopen this thread, please click the button below.",
```

Comment:
I don't think we should allow users to re open

Also you can't click buttons in archived threads